src/common.jl

```diff
@@ -0,0 +1,680 @@
+const XPRS_REFINEOPS_MIPNODELP = 8
+const XPRS_REFINEOPS_LPPRESOLVE = 16
+const XPRS_DUALIZEOPS_SWITCHALGORITHM = 1
+const XPRS_TREEDIAGNOSTICS_MEMORY_USAGE_SUMMARIES = 1
+const XPRS_TREEDIAGNOSTICS_MEMORY_SAVED_REPORTS = 2
+const XPRS_BARPRESOLVEOPS_STANDARD_PRESOLVE = 0
+const XPRS_BARPRESOLVEOPS_EXTRA_BARRIER_PRESOLVE = 1
+const XPRS_PRECOEFELIM_DISABLED = 0
+const XPRS_PRECOEFELIM_AGGRESSIVE = 1
+const XPRS_PRECOEFELIM_CAUTIOUS = 2
+const XPRS_PREDOMROW_AUTOMATIC = -1
+const XPRS_PREDOMROW_DISABLED = 0
+const XPRS_PREDOMROW_CAUTIOUS = 1
+const XPRS_PREDOMROW_MEDIUM = 2
+const XPRS_PREDOMROW_AGGRESSIVE = 3
+const XPRS_PREDOMCOL_AUTOMATIC = -1
+const XPRS_PREDOMCOL_DISABLED = 0
+const XPRS_PREDOMCOL_CAUTIOUS = 1
+const XPRS_PREDOMCOL_AGGRESSIVE = 2
+const XPRS_PRIMALUNSHIFT_ALLOW_DUAL_UNSHIFT = 0
+const XPRS_PRIMALUNSHIFT_NO_DUAL_UNSHIFT = 1
+const XPRS_REPAIRINDEFINITEQ_REPAIR_IF_POSSIBLE = 0
+const XPRS_REPAIRINDEFINITEQ_NO_REPAIR = 1
+const XPRS_OBJ_MINIMIZE = 1.0
+const XPRS_OBJ_MAXIMIZE = -1.0
+const XPRS_TYPE_NOTDEFINED = 0
+const XPRS_TYPE_INT = 1
+const XPRS_TYPE_INT64 = 2
+const XPRS_TYPE_DOUBLE = 3
+const XPRS_TYPE_STRING = 4
+const XPRS_QCONVEXITY_UNKNOWN = -1
+const XPRS_QCONVEXITY_NONCONVEX = 0
+const XPRS_QCONVEXITY_CONVEX = 1
+const XPRS_QCONVEXITY_REPAIRABLE = 2
+const XPRS_QCONVEXITY_CONVEXCONE = 3
+const XPRS_QCONVEXITY_CONECONVERTABLE = 4
+const XPRS_SOLINFO_ABSPRIMALINFEAS = 0
+const XPRS_SOLINFO_RELPRIMALINFEAS = 1
+const XPRS_SOLINFO_ABSDUALINFEAS = 2
+const XPRS_SOLINFO_RELDUALINFEAS = 3
+const XPRS_SOLINFO_MAXMIPFRACTIONAL = 4
+const XPRS_TUNERMODE_AUTOMATIC = -1
+const XPRS_TUNERMODE_OFF = 0
+const XPRS_TUNERMODE_ON = 1
+const XPRS_TUNERMETHOD_AUTOMATIC = -1
+const XPRS_TUNERMETHOD_LPQUICK = 0
+const XPRS_TUNERMETHOD_MIPQUICK = 1
+const XPRS_TUNERMETHOD_MIPCOMPREHENSIVE = 2
+const XPRS_TUNERMETHOD_MIPROOTFOCUS = 3
+const XPRS_TUNERMETHOD_MIPTREEFOCUS = 4
+const XPRS_TUNERMETHOD_MIPSIMPLE = 5
+const XPRS_TUNERMETHOD_SLPQUICK = 6
+const XPRS_TUNERMETHOD_MISLPQUICK = 7
+const XPRS_TUNERMETHOD_MIPHEURISTICS = 8
+const XPRS_TUNERMETHOD_LPNUMERICS = 9
+const XPRS_TUNERTARGET_AUTOMATIC = -1
+const XPRS_TUNERTARGET_TIMEGAP = 0
+const XPRS_TUNERTARGET_TIMEBOUND = 1
+const XPRS_TUNERTARGET_TIMEOBJVAL = 2
+const XPRS_TUNERTARGET_INTEGRAL = 3
+const XPRS_TUNERTARGET_SLPTIME = 4
+const XPRS_TUNERTARGET_SLPOBJVAL = 5
+const XPRS_TUNERTARGET_SLPVALIDATION = 6
+const XPRS_TUNERTARGET_GAP = 7
+const XPRS_TUNERTARGET_BOUND = 8
+const XPRS_TUNERTARGET_OBJVAL = 9
+const XPRS_TUNERHISTORY_IGNORE = 0
+const XPRS_TUNERHISTORY_APPEND = 1
+const XPRS_TUNERHISTORY_REUSE = 2
+const XPRS_TUNERROOTALG_DUAL = 1
+const XPRS_TUNERROOTALG_PRIMAL = 2
+const XPRS_TUNERROOTALG_BARRIER = 4
+const XPRS_TUNERROOTALG_NETWORK = 8
+const XPRS_LPFLAGS_DUAL = 1
+const XPRS_LPFLAGS_PRIMAL = 2
+const XPRS_LPFLAGS_BARRIER = 4
+const XPRS_LPFLAGS_NETWORK = 8
+const XPRS_ISUSERSOLUTION = 0x01
+const XPRS_ISREPROCESSEDUSERSOLUTION = 0x02
```

Comment:
Do we have a Dict from String name to integer value?
This would be very useful for setting attributes via MOI.
No need to implement this now, but we should leave an issue.

Reply:
This exists right now:

https://github.com/JuliaOpt/Xpress.jl/blob/85776d3b22a833068765e2c7ff3aedaec53ccaa0/src/MOI/MOI_wrapper.jl#L367-L370

and can be used as follows:

```julia
MOI.set(model, MOI.RawParameter(Xpress.Lib.XPRS_MPSNAMELENGTH), 64)
```

Is this what you are asking?

Reply:
He's asking for `MOI.set(model, MOI.RawParameter("MPSNAMELENGTH"), 64)`

Reply:
@odow whats you input on that, the constants are ok or we should have strings?
I remember JuMP was planned to unify these with strings.

Reply:
JuMP only accepts strings. So you will not be able to go `set_parameter(model, Xpress.Lib.XPRS_MPSNAMELENGTH, 64)`.

You could do something like:
```Julia
module Foo
    module Lib
       const XPRS_A = 1
       const XPRS_B = 2
       const C = 3
    end
    const STRING_TO_ATTRIBUTE = Dict{String, Any}(
        string(name) => getfield(Lib, name)
        for name in names(Lib; all = true)
        if startswith(string(name), "XPRS_")
    )
end
```

Reply:
Niiiice!